src/App/Search/ReindexFrameworkClient.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+namespace App\Search;
+
+use App\Search\FrameworkSearchClient;
+
+class ReindexFrameworkClient extends AbstractSearchClient
+{
+    /**
+     * @var \App\Search\FrameworkSearchClient
+     */
+    protected $indexClient;
+
+    // update index setting here
+    protected $analysis = [
```

Comment:
Move this into the AbstractSearchClient, as a function which gives the ability to override if required.

Reply:
Something like this?
```
public function overrideFrameworkIndex () 
    {
    // update index setting here
    $analysis = [
        'analysis' => array(
          'analyzer' => array(
            'english_analyzer' => array(
              'tokenizer' => 'standard',
              'filter'    => array('lowercase', 'english_stemmer', 'english_stop'),
            ),
          ),
          'filter'   => array(
            'english_stemmer' => array(
              'type' => 'stemmer',
              'name' => 'english'
            ),
            'english_stop' => array(
              'type' => 'stop',
              'stopwords' => '_english_'
            )
          )
        )
      ];
        $this->getIndexOrCreate();

        $this->reindex($this->getQualifiedIndexName(), $analysis);
    }
```

and then instead of calling `$this->frameworkSearchClient->reindex();` in cli.commands.php I would call `$this->frameworkSearchClient->overrideFrameworkIndex();`

Would that be okay?